numpyro/contrib/autoguide/__init__.py

```diff
@@ -434,12 +430,10 @@ def _get_transform(self, params):
         def loss_fn(z):
             params1 = params.copy()
             params1['{}_loc'.format(self.prefix)] = z
-            # we are doing maximum likelihood, so only require `num_particles=1` and an arbitrary rng_key.
-            return ELBO().loss(random.PRNGKey(0), params1, self.model, self,
-                               *self._args, **self._kwargs)
```

Comment:
We don't store `self._args, self._kwargs` so I move this `loss` function to `setup_prototype` method.